### PR TITLE
fix(connector): [Bluesnap] fix psync status to failure when it is '403' 

### PR DESCRIPTION
**Pull request by srujanchikke**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

When we psync to bluesnap connector when payment gets timed out (status = Processing). Bluesnap send status_code as 403 as connector id is not present in their system . So payment status would be in processing status . This PR is hot-fix for this issue where we are updating status to failure incase of bluesnap . Proper fix for this issue is https://github.com/juspay/hyperswitch-cloud/issues/3064

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

When we psync to bluesnap connector when payment gets timed out (status = Processing). Bluesnap send status_code as 403 as connector id is not present in their system . So payment status would be in processing status . This PR is hot-fix for this issue where we are updating status to failure incase of bluesnap .

## How did you test it?

This can't be tested with exact scenario .Although we can duplicate this scenarion with other case 
1) make a failure payment using failed test scenarios
2)update the status in db to pending
3) psync the payment setting force_synce =true
Before force_sync
Screenshot 2023-11-03 at 5 29 11 PM
After force_sync
Screenshot 2023-11-03 at 1 56 34 PM
## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by jarnura — APPROVED** [review]

File: crates/router/src/core/payments/operations/payment_response.rs [hunk 1]
```diff
@@ -326,7 +326,7 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
                             match err.status_code {
                                 // marking failure for 2xx because this is genuine payment failure
                                 200..=299 => storage::enums::AttemptStatus::Failure,
-                                _ => payment_data.payment_attempt.status,
+                                _ => router_data.status,
```
Comment (jarnura) [inline crates/router/src/core/payments/operations/payment_response.rs:329]: why this change is required?

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/core/payments/operations/payment_response.rs (hunk 1 above)
Comment (srujanchikke) [reply crates/router/src/core/payments/operations/payment_response.rs:329]: Connector doesn't have access to payment_attempt.status , we need to update it using router_data

**Review by jarnura — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jarnura — APPROVED** [review]
